docs/providers.rst

```diff
@@ -121,6 +121,33 @@ App registration (get your key and secret here)
 Development callback URL
     http://localhost:8000/accounts/angellist/login/callback/
 
+Apple
+---------
+
+App registration (create an App ID and then a related Service ID here)
+    https://developer.apple.com/account/resources/certificates/list
+
+Private Key registration (be sure to save it)
+    https://developer.apple.com/account/resources/authkeys/list
+
+Development callback URL
+    http://domain.com/accounts/apple/login/callback/
+
+You'll need to specify Developer's and Application's data:
+
+In Social App:
+    Client ID: Service ID identifier
+    Secret Key: Key ID
+
+In settings.py:
+.. code-block:: python
+
+    SOCIALACCOUNT_PROVIDERS = {
+        'apple': {
+            'MEMBER_ID': 'Member ID/App ID Prefix',
+            'SECRET_KEY' : """-----PRIVATE KEY-----"""
```

Comment:
I think it is better to take the value from the Database like other social applications. You need to increase the `secret_key` field `max_length` for that.

Reply:
I did wonder about that when starting to work on this PR, but I thought that since it's a file, there is actually some benefit in being able to keep it as a file which can be loaded in the settings as needed.

Reply:
Thinking about this more, you are right. 

The auth for the token exchange is based on both the member id and the secret key. There is a field we're not using at the moment `key` which seems like it's used to access the profile API in other clients (e.g. Trello), so perhaps that's a better fit? 

The member_id and the secrey_key are a pair at the moment too - I don't know that it makes sense to have one global and the other per social application. Would you want to see an additional field on the social application model for this?

The final part of this is that  `OAuth2Client` doesn't receive `app.key` at the moment, so I can add that too?

Reply:
Here's an idea of what that would look like https://github.com/0leg5ergeev/django-allauth/pull/5/files (combines your other review comment at the moment)